enhancements/image-references.md

```diff
@@ -0,0 +1,230 @@
+---
+title: image-references-in-operator-bundles
+authors:
+  - "@stevekuznetsov"
+reviewers:
+  - "@jwforres"
+  - "@shawn-hurley"
+  - "@gallettilance"
+  - "@lcarva"
+approvers:
+  - "@jwforres"
+  - "@shawn-hurley"
+creation-date: 2020-05-17
+last-updated: 2020-05-17
+status: implementable
+see-also:
+  - "/enhancements/olm/operator-bundle.md"
+---
+
+# Image References in Operator Bundle Manifests
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Manifests that make up an Operator Bundle for installation via the Operator
+Lifecycle Manager refer to one or more container images by their pull
+specifications: these container images define the operator and operands that
+the manifests deploy and manage. As with OpenShift release payloads, operator
+bundles must refer to images by digest in order to produce reproducible
+installations. A shared process to build operator bundles that replaces image
+references with fully-resolved pull specifications that pin images by digest
+must be built; this process must allow for a number of separate build systems
+to direct how these replacements occur in order to support a full-featured
+build and test strategy.
+
+## Motivation
+
+### Goals
+
+- there is one, canonical, method for building an operator bundle images
+  from a directory of manifests
+- it is possible to perform image resolution and pinning separate from bundle
+  creation, but not the opposite
+- building an operator bundle image does not require the use of a container
+  runtime, elevated privileges or any capacities that are not present for 
+  containerized workloads on OpenShift
+- it is as simple for a developer to build a bundle referring to test versions
+  of operand images as it is for a CI or productized build pipleine to create
+  bundle images for publication
+- operator manifest authors dictate the set of image references in manfiests
+  that must be resolved and pinned
+- operator bundle images may be inspected to determine the pull specifications
+  that were used in the creation of the bundle
+
+### Non-Goals
+
+- operator manifest authors must not be required to define the registry from
```

Comment:
That seems to be a goal, not a `non-goal`

Reply:
How about leading with an explicit "The following statements limit the scope of this enhancement:"?  I'm also fine with "The following are not goals of this enhancement:" and your suggested sense reversal on the list entries.

Reply:
Or lead each entry like "This enhancement will not require operator manifest authors..." or some such.

Reply:
Thats still unclear. `must not ` can par definition not describe a non-goal. A non-goal describes work that should not be done, not work that should be done to avoid a problem. To me its unclear if this here should be done or not.

Reply:
No, it's explicitly a non-goal. The author defines a name. The build system connects that to a location.

Reply:
If its a requirement that the author doesn't define the registry, then this is definitely a goal, not a non-goal.